File: entity-meta-generator-lib/src/main/java/com/loudsight/meta/serialization/transform/IntEntityTransform.java
package com.loudsight.meta.serialization.transform;

import com.loudsight.meta.serialization.EntityTransform;
import com.loudsight.meta.serialization.EntityType;

import java.util.Iterator;
import java.util.List;

public class IntEntityTransform extends EntityTransform<Integer> {

    private static class IntEntityTransformHolder {
        private static final IntEntityTransform INSTANCE = new IntEntityTransform();
    }
    // global access point
    public static IntEntityTransform getInstance() {
        return IntEntityTransform.IntEntityTransformHolder.INSTANCE;
    }
    private IntEntityTransform() {
        super(EntityType.INTEGER, Integer.class);
    }


    @Override public void serializeEntity( Integer entity, List<Byte> bytes) {
        bytes.add(EntityType.INTEGER.getCode());
        writeInt(entity, bytes);
    }

    @Override public Integer  deserializeEntity(Iterator<Byte> bytes) {
        return readInt(bytes);
    }
}

File: entity-meta-generator-lib/src/main/kotlin/com/loudsight/meta/serialization/transform/IntEntityTransform.kt
package com.loudsight.meta.serialization.transform

import com.loudsight.meta.serialization.EntityTransform
import com.loudsight.meta.serialization.EntityType

object IntEntityTransform : EntityTransform<Int>(EntityType.INTEGER, Int::class) {

    override fun serializeEntity(entity: Int, bytes: MutableList<Byte>) {
        bytes.add(EntityType.INTEGER.code)
        writeInt(entity, bytes)
    }

    override fun deserializeEntity(bytes: Iterator<Byte>): Int {
        return readInt(bytes)
    }


    override fun canTransform(entity: Any): Boolean {
        return entity is Int
    }
}